### PR TITLE
Make polkit optional for a minimal gui system

### PR DESCRIPTION
**Pull request by acertain**
Should I make a global `config.polkit` option instead of per-package options?

### CODE REVIEW

File: nixos/modules/services/x11/xserver.nix [hunk 1]
```diff
@@ -434,9 +434,6 @@ in
               choose between ‘programs.ssh.startAgent’ and ‘services.xserver.startGnuPGAgent’.
             '';
         }
-        { assertion = config.security.polkit.enable;
```
Comment (fread2281) [inline nixos/modules/services/x11/xserver.nix]: Why was this there?

**Comment by vcunat** [comment]
I don't understand the motivation. Is it disk usage? I'd think removing polkit from networkmanager and/or gnome makes a rather small change on the closure size. Just e.g. removing documentation from closure would likely have much larger impact.

**Comment by edolstra** [comment]
I'm kinda :-1: on this because it just makes it harder to test NixOS (we'd have to check whether certain features work for the polkit and non-polkit case).

**Comment by lethalman** [comment]
Write down a rationale please.

**Comment by fread2281** [comment]
The rationale is that `security.polkit.enable` already exists and a javascipt vm (which polkit depends on) is a big dependency for a core system component (in both build time and disk usage) and a rather large security surface.

**Comment by vcunat** [comment]
This inspired me to easily cut a few MBs from it 36194332e (I'll push it to staging soon).

**Comment by lethalman** [comment]
@fread2281 turns out that gnome-shell uses javascript, so there's still a core component that it's using. So you didn't save anything at all.

About large security surface, polkit itself probably doesn't have a vulnerability since 1 year.

Anyway, apart the missing rationale, I will review this. I'm not opposed to adding optional stuff if people is willing to use it, it's written nowhere everything should be tested by everyone. Hence it's easy, "I'm not going to maintain this gnome stuff without polkit", problem solved.

- [ ] In your PR polkit is disabled by default. Please enable it by default by replacing all the `or false` with `or true`.
- [ ] The NM plugins should be aware of the fact that NM has been compiled with polkit support, without telling that explicitly everytime which is very error prone. Please add `passthru.polkitSupport = true;` to networkmanager and use that in the plugins.

**Comment by vcunat** [comment]
@lethalman, ``spidermonkey_17`` is used only for polkit and nothing else (in current master).

**Comment by lethalman** [comment]
@vcunat ah that different js engine... so 18mb less for a gnome desktop.

**Comment by edolstra** [comment]
Can't it use the regular spidermonkey?

**Comment by vcunat** [comment]
@edolstra: it fails to configure, searching for 185 or 17. The only newer one is 24, but I see even Arch uses polkit with 17. (Our default ``spidermonkey`` attribute is even older than all these.)

**Comment by fread2281** [comment]
@lethalman done. The plugins actually don't need polkit. Note that setting `security.polkit.enable = false` will set `config.polkit = false`.

**Comment by fread2281** [comment]
The other rationale for this is that spidermonkey is a relatively long build for people that can't use hydra's cache.

**Comment by lethalman** [comment]
@fread2281 disabling polkit support in packages when disabling the polkit service is not a good idea. First of all, it will recompile those packages, and I'm sure it can be worse than just recompiling spidermonkey as you say. The choice must be up to the user to disable polkit support in packages.

File: nixos/modules/services/x11/xserver.nix (hunk 1 above)
Comment (lethalman) [inline nixos/modules/services/x11/xserver.nix]: You can use `git blame`: 8e1f243b309f31f3c6605cc7bad17dbd1ccad568

**Comment by lethalman** [comment]
So apart fixing the assertion and not taking initiative on nixpkgs config, it's :+1: to me. Will test it afterwards.

**Comment by edolstra** [comment]
I'm still :-1: on this due to the unnecessary complexity it introduces. It would force every service that uses polkit to consider two different configurations and provides alternatives for either situation. Especially for something security-sensitive, I don't think that's not a good idea.

(Yes, we should also get rid of `security.polkit.enable`.)

**Comment by lethalman** [comment]
@edolstra what services are you talking about? I only see NM, in which case it will be asserted it won't work without polkit.

**Comment by fread2281** [comment]
NM and systemd work without polkit. NM allows all afaict, and systemd 
only allows root.

**Comment by edolstra** [comment]
Right, I was thinking about systemd, which uses polkit. The fact that it works without polkit is irrelevant, it still doubles the test matrix to support systemd without polkit.

**Comment by lethalman** [comment]
@edolstra we can also point out in the description that disabling polkit is not well supported/tested and that's it, yet allow people to do it and drop our responsibility.

**Comment by fread2281** [comment]
@lethalman updated

**Comment by joachifm** [comment]
Is this still being worked on or has it been abandoned?

**Comment by joachifm** [comment]
No response for months, closing as abandoned.

**Pull request closed without merging**